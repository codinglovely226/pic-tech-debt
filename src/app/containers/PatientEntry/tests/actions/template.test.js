import {
  updateTemplatePatientAge,
  updateTemplatePatientAgeGroup,
  updateTemplatePatientAgeYears,
  updateTemplatePatientAnimalType,
  updateTemplatePatientCategory,
  updateTemplatePatientCircumstance,
  updateTemplatePatientLocation,
  updateTemplatePatientMedicalRecordNumber,
  updateTemplatePatientName,
  updateTemplatePatientOccupation,
  updateTemplatePatientPoisonSeverityScore,
  updateTemplatePatientPlaceId,
  updateTemplatePatientAddress1,
  updateTemplatePatientAddress2,
  updateTemplatePatientSuburb,
  updateTemplatePatientState,
  updateTemplatePatientPostcode,
  updateTemplatePatientCountry,
  updateTemplatePatientISOCountry,
  updateTemplatePatientTelephone,
  updateTemplatePatientGeometry,
  updateTemplatePatientPregnantLactating,
  updateTemplatePatientPregnantStatus,
  updateTemplatePatientPregnantTrimester,
  updateTemplatePatientPregnantTrimesterUnknown,
  updateTemplatePatientRiskAssessment,
  updateTemplatePatientSex,
  updateTemplatePatientWeight,
} from '../../actions/template';

describe('Patient action creators', () => {
  it('updateTemplatePatientAge', () => {
    expect(updateTemplatePatientAge(1, 10, 20, 'months', 'years')).toMatchSnapshot();
  });
  it('updateTemplatePatientAgeGroup', () => {
    expect(updateTemplatePatientAgeGroup(1, 'months', 'years')).toMatchSnapshot();
  });
  it('updateTemplatePatientAgeYears', () => {
    expect(updateTemplatePatientAgeYears(1, '', '0.75')).toMatchSnapshot();
  });
  it('updateTemplatePatientAnimalType', () => {
    expect(updateTemplatePatientAnimalType(1, 'dog', 'bird')).toMatchSnapshot();
  });
  it('updateTemplatePatientCategory', () => {
    expect(updateTemplatePatientCategory(1, 'animal', 'human')).toMatchSnapshot();
  });
  it('updateTemplatePatientCircumstance', () => {
    expect(updateTemplatePatientCircumstance(1, 'Old Circumstance', 'New Circumstance')).toMatchSnapshot();
  });
  it('updateTemplatePatientLocation', () => {
    expect(updateTemplatePatientLocation(1, 'Old Location', 'New Location')).toMatchSnapshot();
  });
  it('updateTemplatePatientMedicalRecordNumber', () => {
    expect(updateTemplatePatientMedicalRecordNumber(1, '1234', '5678')).toMatchSnapshot();
  });
  it('updateTemplatePatientName', () => {
    expect(updateTemplatePatientName(1, 'Old Name', 'New Name')).toMatchSnapshot();
  });
  it('updateTemplatePatientOccupation', () => {
    expect(updateTemplatePatientOccupation(1, 'Old Occupation', 'New Occupation')).toMatchSnapshot();
  });
  it('updateTemplatePatientPoisonSeverityScore', () => {
    expect(updateTemplatePatientPoisonSeverityScore(1, 'Minimal', 'Minor')).toMatchSnapshot();
  });
  it('updateTemplatePatientPlaceId', () => {
    expect(updateTemplatePatientPlaceId(1, 'Old Place Id', 'New Place Id')).toMatchSnapshot();
  });
  it('updateTemplatePatientAddress1', () => {
    expect(updateTemplatePatientAddress1(1, 'Old Address1', 'New Address 1')).toMatchSnapshot();
  });
  it('updateTemplatePatientAddress2', () => {
    expect(updateTemplatePatientAddress2(1, 'Old Address1', 'New Address 1')).toMatchSnapshot();
  });
  it('updateTemplatePatientSuburb', () => {
    expect(updateTemplatePatientSuburb(1, 'Perth', 'Sydney')).toMatchSnapshot();
  });
  it('updateTemplatePatientState', () => {
    expect(updateTemplatePatientState(1, 'WA', 'NSW')).toMatchSnapshot();
  });
  it('updateTemplatePatientPostcode', () => {
    expect(updateTemplatePatientPostcode(1, '6000', '3000')).toMatchSnapshot();
  });
  it('updateTemplatePatientCountry', () => {
    expect(updateTemplatePatientCountry(1, 'Australia', 'New Zealand')).toMatchSnapshot();
  });
  it('updateTemplatePatientISOCountry', () => {
    expect(updateTemplatePatientISOCountry(1, 'AU', 'NZ')).toMatchSnapshot();
  });
  it('updateTemplatePatientTelephone', () => {
    expect(updateTemplatePatientTelephone(1, '123456', '56789')).toMatchSnapshot();
  });
  it('updateTemplatePatientGeometry', () => {
    expect(updateTemplatePatientGeometry(1, {}, {})).toMatchSnapshot();
  });
  it('updateTemplatePatientPregnantLactating', () => {
    expect(updateTemplatePatientPregnantLactating(1, 'No', 'Yes')).toMatchSnapshot();
  });
  it('updateTemplatePatientPregnantStatus', () => {
    expect(updateTemplatePatientPregnantStatus(1, 'No', 'Yes')).toMatchSnapshot();
  });
  it('updateTemplatePatientPregnantTrimester', () => {
    expect(updateTemplatePatientPregnantTrimester(1, '1', '2')).toMatchSnapshot();
  });
  it('updateTemplatePatientPregnantTrimesterUnknown', () => {
    expect(updateTemplatePatientPregnantTrimesterUnknown(1, false, true)).toMatchSnapshot();
  });
  it('updateTemplatePatientRiskAssessment', () => {
    expect(updateTemplatePatientRiskAssessment(1, 'Old Assessment', 'New Assessment')).toMatchSnapshot();
  });
  it('updateTemplatePatientSex', () => {
    expect(updateTemplatePatientSex(1, 'male', 'female')).toMatchSnapshot();
  });
  it('updateTemplatePatientWeight', () => {
    expect(updateTemplatePatientWeight(1, '80', '83', 'kg', 'kg')).toMatchSnapshot();
  });
});
