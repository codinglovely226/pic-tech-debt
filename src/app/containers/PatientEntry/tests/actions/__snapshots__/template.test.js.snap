exports[`Patient action creators updateTemplatePatientAddress1 1`] = `
Object {
  "address1": "New Address 1",
  "id": 1,
  "old": "Old Address1",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_ADDRESS1",
}
`;

exports[`Patient action creators updateTemplatePatientAddress2 1`] = `
Object {
  "address2": "New Address 1",
  "id": 1,
  "old": "Old Address1",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_ADDRESS2",
}
`;

exports[`Patient action creators updateTemplatePatientAge 1`] = `
Object {
  "age": 20,
  "ageUnits": "years",
  "oldAge": "",
  "oldAgeUnits": "",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_AGE",
}
`;

exports[`Patient action creators updateTemplatePatientAgeGroup 1`] = `
Object {
  "ageGroup": "years",
  "old": "months",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_AGE_GROUP",
}
`;

exports[`Patient action creators updateTemplatePatientAgeYears 1`] = `
Object {
  "ageYears": "0.75",
  "old": "",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_AGE_YEARS",
}
`;

exports[`Patient action creators updateTemplatePatientAnimalType 1`] = `
Object {
  "animalType": "bird",
  "old": "dog",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_ANIMAL_TYPE",
}
`;

exports[`Patient action creators updateTemplatePatientCategory 1`] = `
Object {
  "category": "human",
  "old": "animal",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_CATEGORY",
}
`;

exports[`Patient action creators updateTemplatePatientCircumstance 1`] = `
Object {
  "circumstance": "New Circumstance",
  "old": "Old Circumstance",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_CIRCUMSTANCE",
}
`;

exports[`Patient action creators updateTemplatePatientCountry 1`] = `
Object {
  "country": "New Zealand",
  "id": 1,
  "old": "Australia",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_COUNTRY",
}
`;

exports[`Patient action creators updateTemplatePatientGeometry 1`] = `
Object {
  "geometry": Object {},
  "old": Object {},
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_GEOMETRY",
}
`;

exports[`Patient action creators updateTemplatePatientISOCountry 1`] = `
Object {
  "id": 1,
  "isoCountry": "NZ",
  "old": "AU",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_ISOCOUNTRY",
}
`;

exports[`Patient action creators updateTemplatePatientLocation 1`] = `
Object {
  "location": "New Location",
  "old": "Old Location",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_LOCATION",
}
`;

exports[`Patient action creators updateTemplatePatientMedicalRecordNumber 1`] = `
Object {
  "appKey": undefined,
  "medicalRecordNumber": "5678",
  "old": "1234",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_MEDICAL_RECORD_NUMBER",
}
`;

exports[`Patient action creators updateTemplatePatientName 1`] = `
Object {
  "appKey": undefined,
  "name": "New Name",
  "old": "Old Name",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_NAME",
}
`;

exports[`Patient action creators updateTemplatePatientOccupation 1`] = `
Object {
  "occupation": "New Occupation",
  "old": "Old Occupation",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_OCCUPATION",
}
`;

exports[`Patient action creators updateTemplatePatientPlaceId 1`] = `
Object {
  "id": 1,
  "old": "Old Place Id",
  "placeId": "New Place Id",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_PLACEID",
}
`;

exports[`Patient action creators updateTemplatePatientPoisonSeverityScore 1`] = `
Object {
  "old": "Minimal",
  "poisonSeverityScore": "Minor",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_POISON_SEVERITY_SCORE",
}
`;

exports[`Patient action creators updateTemplatePatientPostcode 1`] = `
Object {
  "old": "6000",
  "postcode": "3000",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_POSTCODE",
}
`;

exports[`Patient action creators updateTemplatePatientPregnantLactating 1`] = `
Object {
  "old": "No",
  "pregnantLactating": "Yes",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_PREGNANT_LACTATING",
}
`;

exports[`Patient action creators updateTemplatePatientPregnantStatus 1`] = `
Object {
  "old": "No",
  "pregnantStatus": "Yes",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_PREGNANT_STATUS",
}
`;

exports[`Patient action creators updateTemplatePatientPregnantTrimester 1`] = `
Object {
  "old": "1",
  "pregnantTrimester": "2",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_PREGNANT_TRIMESTER",
}
`;

exports[`Patient action creators updateTemplatePatientPregnantTrimesterUnknown 1`] = `
Object {
  "old": false,
  "pregnantTrimesterUnknown": true,
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_PREGNANT_TRIMESTER_UNKNOWN",
}
`;

exports[`Patient action creators updateTemplatePatientRiskAssessment 1`] = `
Object {
  "old": "Old Assessment",
  "riskAssessment": "New Assessment",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_RISK_ASSESSMENT",
}
`;

exports[`Patient action creators updateTemplatePatientSex 1`] = `
Object {
  "gender": "female",
  "old": "male",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_SEX",
}
`;

exports[`Patient action creators updateTemplatePatientState 1`] = `
Object {
  "id": 1,
  "old": "WA",
  "state": "NSW",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_STATE",
}
`;

exports[`Patient action creators updateTemplatePatientSuburb 1`] = `
Object {
  "old": "Perth",
  "suburb": "Sydney",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_SUBURB",
}
`;

exports[`Patient action creators updateTemplatePatientTelephone 1`] = `
Object {
  "id": 1,
  "old": "123456",
  "telephone": "56789",
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_TELEPHONE",
}
`;

exports[`Patient action creators updateTemplatePatientWeight 1`] = `
Object {
  "oldWeight": "",
  "oldWeightUnits": "",
  "templateId": 1,
  "type": "pic/PatientEntry/UPDATE_TEMPLATE_PATIENT_WEIGHT",
  "weight": "83",
  "weightUnits": "kg",
}
`;
